source/common/http/conn_manager_utility.cc

```diff
@@ -131,6 +131,52 @@ void ConnectionManagerUtility::mutateRequestHeaders(Http::HeaderMap& request_hea
   if (config.tracingConfig()) {
     Tracing::HttpTracerUtility::mutateHeaders(request_headers, runtime);
   }
+
+  if ((connection.ssl() && connection.ssl()->isMtls()) ||
+      config.forwardClientCert() == Http::ForwardClientCertType::AlwaysForwardOnly) {
+    std::string clientCertDetails;
```

Comment:
Quick related drive by comment (not full review): Can we pull all of this code out of mutateRequestHeaders() into a new function that can be self contained tested? This function is getting too big as it is.